src/tracing/builder/geth.rs

```diff
@@ -142,15 +142,16 @@ impl GethTraceBuilder {
         call_frames.push((0, root_call_frame));
 
         for (idx, trace) in self.nodes.iter().enumerate().skip(1) {
+            // include logs only if call and all its parents were successful
+            let include_logs = include_logs && !self.call_or_parent_failed(trace);
+            call_frames.push((idx, trace.geth_empty_call_frame(include_logs)));
+
             // selfdestructs are not recorded as individual call traces but are derived from
-            // the call trace and are added as additional `CallFrame` objects to the parent call
+            // the call trace and are added as additional `CallFrame` objects
+            // becoming the first child of the derived call
```

Comment:
ah I see, this adds the call frame first and then adds the sefldestruct to that callframe that we just added as a child call